internal/sanitize.go

```diff
@@ -27,16 +27,17 @@ func Sanitize(s string) string {
 	if len(s) == 0 {
 		return s
 	}
+	if len(s) > labelKeySizeLimit {
```

Comment:
This truncation still needs to happen last, because s might grow by 3 or 4 characters below.